src/include/migraphx/shape.hpp

```diff
@@ -156,8 +156,33 @@ struct shape
 
     shape(const std::vector<shape>& subs);
 
+    /**
+     * Creates an output shape with dimensions equal to the input lengths and strides determined
+     * by the permutation argument.
+     * The "original" dimensions of the shape can be considered to be the forward permutation of the
+     * given output dimensions.
+     * The output shape's strides are the inverse permutation of the standard strides generated by
+     * the "original" dimensions.
+     *
+     * 2D example:
+     *   parameters:
+     *     l = [2, 3], perm = [1, 0]
+     *   therefore:
+     *     inverse_perm = [1, 0]
```

Comment:
```suggestion
     *     inverse_perm = [0, 1]
```

Reply:
No, the inverse permutation is [1, 0]. It needs to swap it back.

Reply:
Just remove this.